CONTRIBUTING_GUIDE.md

```diff
@@ -0,0 +1,209 @@
+1. When picking up an issue give a brief presentation about yourself.
+
+```
+Template:
+Hi, I'm [Your Name] and I'll be working on issue #[Issue Number].
+
+I estimate this will take [Time Estimate] to complete.
+
+This is how I would tackle this issue:
+[Steps to solve issue]
+
+```
+
+2. **Estimated Time to Completion**: Approximate number of hours or days required to complete the task.
+
+3. **Approach and Methodology**: Briefly outline your strategy for tackling the issue, including any relevant tools, technologies, or resources you plan to utilize.
+
+# **🎯 Pull Request**
+
+1. **Clone and Fork Repo**: Click the **Fork** button in the top-right corner to create a copy of the repository under your account.
+
+    - <a href="https://github.com/ZencypherSolutions/semaphore-stellar-sdk" target="_blank"> HERE</a>
+
+#
+
+2. **Clone the Fork:** 
+    - Clone the forked repository to your local machine by running the following command:
+
+    ```bash
+   git clone https://github.com/YOUR_USERNAME/REPOSITORY_NAME.git
+   ```
+
+    - Replace `YOUR_USERNAME` and `REPOSITORY_NAME` with your GitHub username and the repository name.
+
+#
+
+3. **Create a new branch or use the main branch:** When modifying contracts kindly make sure the formatting is correct and all tests pass successfully.
+
+    - Create a branch name based on the type of change (e.g., `feat/name-related-issue`, `docs/name-related-issue`).
+
+    ```
+    git checkout -b branch-name
+    ```
+    - One of ideas on how to implement it for the branch name:
+
+        > `docs/update-readme` or `fix/bottom-bug`.
+
+#
+
+4. **Commit:** Commit your changes.
+
+    1. **git add (file-name)**
+    2. **git commit -m "[type] description"**
+
+    - Example: 
+    ```
+    `git add Create_Documentation`
+
+    `git commit -m "[docs]: update documentation"`
+    ```
+
+#
+
+5. **Push fork:** Push to your fork and submit a pull request on our `dev` branch. Please provide us with some explanation of why you made the changes you made. For new features make sure to explain a standard use case to us.
+
+- Push your changes to your forked repository:
+    ```bash
+   git push origin your-branch-name
+   ```
+   > Replace `your-branch-name` with the name of your branch.
+
+- Example: 
+
+    ```bash
+    `git push origin fix/bug-fix`
+    ```
+
+#
+
+6. **Submit a Pull Request:** Submit a pull request to the `dev` branch of the Semaphore Stellar SDK repository.
+
+    - <a href="https://github.com/ZencypherSolutions/semaphore-stellar-sdk/compare" target="_blank"> Summit pull request</a>
+
+
+
+# **📝 CI (Github Actions)**
+
+We use GitHub Actions to verify if the code of your PR passes all our checks.
+
+When you submit your PR (or later change that code), a CI build will automatically be kicked off. A note will be added to the PR, and will indicate the current status of the build.
+
+# **📁 Commits**
+
+You can do a regular commit by following the next:
+
+``` [type] significant message ```
+
+### Type
```

Comment:
This looks a lot like the conventional commits. It would be a good idea to add a reference to the website, so that people can learn more about it.

https://www.conventionalcommits.org/en/v1.0.0/